staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -946,10 +946,17 @@ func decode(codec runtime.Codec, versioner storage.Versioner, value []byte, objP
 	if _, err := conversion.EnforcePtr(objPtr); err != nil {
 		return fmt.Errorf("unable to convert output object to pointer: %v", err)
 	}
-	_, _, err := codec.Decode(value, nil, objPtr)
```

Comment:
so every list request that returns N objects will increment this metric N times? that seems like a lot of metrics updates per API request. It also makes it hard to understand the meaning of the metric if the counts aren't related to the actual number of items stored per gvk

Reply:
> so every list request that returns N objects will increment this metric N times?

Yeah， Considering that object‘s coding may vary even in a signal list request (eg: deployment objects in apps/v1 and apps/v1beta )

>  It also makes it hard to understand the meaning of the metric if the counts aren't related to the actual number of items stored per gvk

By exposing this metric, we can tell into which gvk stored object decoded and which deprecated gvk still be in use , these object in deprecated gvk coding should be migrated to the latest gvk.